server/src/error.rs

```diff
@@ -0,0 +1,26 @@
+use core::{
+    error::Error,
+    fmt::{Debug, Display},
+};
+
+use crate::platform::{crypto::Drbg, TpmContextDeps};
+
+pub enum ServerError<Deps: TpmContextDeps> {
+    Drbg(<Deps::Drbg as Drbg>::Error),
```

Comment:
I'm still not sure I'm on-board with everyone bringing their own error type into the server library. At some point the error is either discarded and we enter failure mode, or it's converted to a `TpmReturnCode`. So no matter what this error type gets squeezed down into one of two distinct possible other types. That smells an awful lot like us not needing to know the exact error type, and that we'd provide some error type that the client must convert into - leaving the ball in their court for how to do that conversion.

Like this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=12ab50e115eddc5f8658e15f99c4c727